set.mm

```diff
@@ -99162,6 +99162,134 @@ a Cantor normal form (and injectivity, together with coherence
+    frrlem16 $p |- ( ( ( R Fr A /\ R Se A ) /\ z e. A ) ->
+       A. w e. TrPred ( R , A , z )
+       Pred ( R , A , w ) C_ TrPred ( R , A , z ) ) $=
+      ( wfr wse wa cv wcel cpred ctrpred wss simplr simpllr simpr trpredtr sylc
+      jca ralrimiva ) CDEZCDFZGZAHZCIZGZCDBHZJCDUCKZLZBUGUEUFUGIZGZUDUAGUIUHUJU
+      DUAUBUDUIMTUAUDUINRUEUIOCDUCUFPQS $.
+  $}
+
+  ${
+    $d F x y z u v a b c f g h $.  $d R x y z u v a b c f g h $.
+    $d A x y z u v a b c f g h $.  $d G x y z u v a b c f g h $.
+    frr.1 $e |- F = frecs ( R , A , G ) $.
+    $( Law of general well-founded recursion, part one.  This theorem and the
+       following two drop the partial order requirement from ~ fpr1 , ~ fpr2 ,
+       and ~ fpr3 , which requires using the axiom of infinity (Contributed by
+       Scott Fenton, 11-Sep-2023.) $)
+    frr1 $p |- ( ( R Fr A /\ R Se A ) -> F Fn A ) $=
+      ( vx vy vv vu va vb vc vf vg vh vz wa cv wss cvv wfr wfun wceq cpred wral
+      wse cdm wfn cfv cres co w3a wex cab eqid frrlem1 frrlem15 frrlem9 ctrpred
+      cop csn cun simpl wcel setlikespec ancoms adantll trpredpred syl frrlem16
+      trpredex a1i trpredss cdif wne difss frmin mpanr1 frrlem14 df-fn sylanbrc
+      c0 wrex ) ABUAZABUFZQZCUBCUGZAUCCAUHWFFGHIAJRZKRZUHWIASABLRZUDZWISLWIUEQW
+      JWHUIWJWHWKUJDUKUCLWIUEULKUMJUNZBMNOCDKLFGAWLBJMDWLUOUPZEFGHIAWLBMNOCDWME
+      UQZURWFFGPJHIAWLCABPRZUSZUJWOWOCABWOUDZUJDUKUTVAVBZBWPMNOCDWMEWNWRUOWDWEV
+      CWFWOAVDZQZWQTVDZWQWPSWEWSXAWDWSWEXAABWOVEVFVGZATBWOVHVIPJABVJWPTVDWTABWO
+      VKVLWTXAWPASXBATBWOVMVIWFAWGVNZASXCWBVOXCBWOUDWBUCPXCWCAWGVPPAXCBVQVRVSCA
+      VTWA $.
+
+    $d X x y z u v a b c f g h $.
+    $( Law of general well-founded recursion, part two.  Now we establish the
+       value of ` F ` within ` A ` .  (Contributed by Scott Fenton,
+       11-Sep-2023.) $)
+    frr2 $p |- ( ( ( R Fr A /\ R Se A ) /\ X e. A ) ->
+      ( F ` X ) = ( X G ( F |` Pred ( R , A , X ) ) ) ) $=
+      ( vy vx va vb vc vf wa wcel cfv cpred cres co wceq cv vv vu vg vh wfr wse
+      cdm frr1 fndmd eleq2d pm5.32i wi fveq2 id predeq3 reseq2d oveq12d eqeq12d
+      imbi2d wfn wss wral w3a wex eqid frrlem1 frrlem15 frrlem10 expcom vtoclga
+      cab impcom sylbir ) ABUEABUFMZEANZMVNECUGZNZMECOZECABEPZQZDRZSZVNVQVOVNVP
+      AEVNACABCDFUHUIUJUKVQVNWBVNGTZCOZWCCABWCPZQZDRZSZULVNWBULGEVPWCESZWHWBVNW
+      IWDVRWGWAWCECUMWIWCEWFVTDWIUNWIWEVSCABWCEUOUPUQURUSVNWCVPNWHVNHGUAUBAITZJ
+      TZUTWKAVAABKTZPZWKVAKWKVBMWLWJOWLWJWMQDRSKWKVBVCJVDIVKZBLUCUDCDJKHGAWNBIL
+      DWNVEVFZFHGUAUBAWNBLUCUDCDWOFVGVHVIVJVLVM $.
+
+    $d H z $.
+    $( Law of general well-founded recursion, part three.  Finally, we show
+       that ` F ` is unique.  We do this by showing that any function ` H `
+       with the same properties we proved of ` F ` in ~ frr1 and ~ frr2 is
+       identical to ` F ` .  (Contributed by Scott Fenton, 11-Sep-2023.) $)
+    frr3 $p |- ( ( ( R Fr A /\ R Se A ) /\ ( H Fn A /\
```

Comment:
It looks like I forgot to "validate" the following comment, sorry:

It looks like frr3g is merely a lemma for frr3 (and another lemma) ?  Maybe it could be labeled and commented as such.

Reply:
It's named that way by analogy with fpr3g. It's a little more general statement than frr3 and I figured it was interesting enough to stand on its own.

Reply:
I had missed that one. I'd say that fpr3g is similarly a lemma. Why do you think it is more general ? Once we have frr3, then we know that all functions with the given characteristic property are equal to the one defined by wrecs, which seems to be sufficient.  If we wanted, we could recover frr3g by one application of transitivity of equality.

Reply:
If we are talking about recovering it, why not just use it in the first place? It's interesting to show that it doesn't depend on the rest of the development and it's going to be there in either case.

Reply:
My remark about the ease of recovering it in the hypothesis that we would want to use it was to show that it added nothing. But, ok, let's have this merged. Resolving this conversation.